typed-function.js

```diff
@@ -606,7 +608,7 @@
           '" do not match any of the defined signatures of function ' + _name + '.');
       err.data = {
         category: 'mismatch',
-        actual: args.map(findTypeName)
+        actual: flatMap(args, arg => Object.keys(findTypeNames(arg)))
       }
```

Comment:
Similarly, here I'd suggest joining the keys of findTypeNames(arg) with ' & ' (and then maybe it can be left as map rather than flatMap? I'm not 100% clear on what flatMap does). (In fact, one could still have a `findTypeName()` function which just joins the keys of `findTypeNames()` with ' & '.)

Reply:
Good point. In the errors, we put the "raw" information in the `.data` property. In this case, `actual` should contain a list with the type of each argument. But now, we can have multiple types per argument. So that is an array containing an array. The `flatMap` doesn't make sense here, it should respect the number of arguments. We could stringify each argument, but since this is about providing "raw" information I think it makes sense to keep it as an array containing arrays, so you can do formatting of that yourself if you want. Does that make sense?

Reply:
Sure, I mean to the extent anyone is using the error.data, having each entry already split into an array is probably easier to use, as long as the semantics are clear.  (There is a slight bigger-picture difficulty now emerging here in that in some places an array of type names represents the union of those types and in other places it represents the intersection of those types and it can get a bit confusing. For sanity's sake, it might be worth shifting to a data structure that explicitly keeps track of unions or of intersections of types -- I mean so that you could query the object as to which it was standing for.)

Reply:
👍  will keep it like this for now.

Yeah, the `error.data` format is a bit messy and inconsistent. That could use an overhaul, but let's park that for a separate PR.